script/tool/test/util.dart

```diff
@@ -110,45 +120,38 @@ void createFakeCHANGELOG(Directory parent, String texts) {
 }
 
 /// Creates a `pubspec.yaml` file with a flutter dependency.
+///
+/// [platformSupport] is a map of platform string to the support details for
+/// that platform. If empty, no `plugin` entry will be created unless `isPlugin`
+/// is set to `true`.
 void createFakePubspec(
   Directory parent, {
   String name = 'fake_package',
   bool isFlutter = true,
-  PlatformSupport? androidSupport,
-  PlatformSupport? iosSupport,
-  PlatformSupport? linuxSupport,
-  PlatformSupport? macosSupport,
-  PlatformSupport? webSupport,
-  PlatformSupport? windowsSupport,
+  bool isPlugin = false,
+  Map<String, PlatformSupport> platformSupport =
+      const <String, PlatformSupport>{},
   String publishTo = 'http://no_pub_server.com',
   String? version,
 }) {
+  isPlugin |= platformSupport.isNotEmpty;
   parent.childFile('pubspec.yaml').createSync();
   String yaml = '''
 name: $name
+''';
+  if (isFlutter) {
+    if (isPlugin) {
+      yaml += '''
```

Comment:
nit: Instead of manipulating strings, we could create the datastructure then use a yaml library to generate the yaml.  That would make this a bit more maintainable.

Reply:
Yes, I've thought about that as well. Initial investigation turned up mostly libraries that can read YAML, unfortunately, rather than write it, but I'd like to look into it more at some point.